src/network/C4Network2IO.cpp

```diff
@@ -1264,15 +1292,23 @@ void C4Network2IO::SendConnPackets()
 
 void C4Network2IO::OnPuncherConnect(C4NetIO::addr_t addr)
 {
-	// Sanity check
-	if (addr.GetFamily() != C4NetIO::HostAddress::IPv4)
-		return;
-	Application.InteractiveThread.ThreadLogS("Adding address from puncher: %s", addr.ToString().getData());
+	// NAT punching is only relevant for IPv4, so convert here to show a proper address.
```

Comment:
Does the IPv6 (firewall hole) puncher not connect back to the client then? Should there be an assertion (& early bail in release builds) to ensure this?